account-kit/signer/src/base.ts

```diff
@@ -1520,20 +1520,41 @@ function subscribeWithDelayedFireImmediately<T>(
   selector: (state: AlchemySignerStore) => T,
   listener: (selectedState: T, previousSelectedState: T) => void,
 ): () => void {
-  let subscribeHasReturned = false;
+  // Until this function has returned, maintain a queue of events that will be
+  // processed once it does.
+  const queue: [selectedState: T, previousSelectedState: T][] = [];
+  let queueHasEmptied = false;
+  let hasUnsubscribed = false;
+
+  function drainQueue() {
+    // Note that this handles the case where more events are added to the queue
+    // during the handling of previous events.
+    for (let i = 0; i < queue.length && !hasUnsubscribed; i++) {
+      const args = queue[i];
+      listener(...args);
+    }
+    queueHasEmptied = true;
+  }
+
+  // Schedule the queue to be drained after this function has returned.
+  setTimeout(drainQueue, 0);
```

Comment:
do you think it might be worth cancelling the timeout in the function returned by this? or is it impossible for that to fire first?

Reply:
makes sense!

Reply:
i am going to merge this now, thanks